File: client/src/Card.ts
import { Colour, colourToString, Fill, fillToString, Quantity, Shape, shapeToString } from "./card_enums";

export class Card {

    public image_path: string;
    public id: number;


    public quantity: Quantity;
    public colour: Colour;
    public shape: Shape;
    public fill: Fill;

    constructor(quantity: Quantity, colour: Colour, shape: Shape, fill: Fill) {
        
        this.quantity = quantity;
        this.colour = colour;
        this.shape = shape;
        this.fill = fill;

        // determined from factors
        this.image_path = "/sets_shapes/" + shapeToString(shape) + "_" + colourToString(colour) + "_" + fillToString(fill) + ".png";
        this.id = quantity * 27 + colour * 9 + shape * 3 + fill * 1 + 1;

        console.log(this.id)

    }
}

File: server/card.py
from card_enums import Colour, Fill, Quantity, Shape, colour_to_string, fill_to_string, shape_to_string


class Card:
    def __init__(self, quantity: Quantity, colour: Colour, shape: Shape, fill: Fill):
        self.quantity = quantity
        self.colour = colour
        self.shape = shape
        self.fill = fill

        self.image_path = f"/sets_shapes/{shape_to_string(shape)}_{colour_to_string(colour)}_{fill_to_string(fill)}.png"
        self.id = quantity.value * 27 + colour.value * 9 + shape.value * 3 + fill.value * 1 + 1

        #print(self.id)

    def to_dict(self):
        return {
            "id": self.id,
            "quantity": self.quantity.value,
            "colour": self.colour.value,
            "shape": self.shape.value,
            "fill": self.fill.value,
            "image_path": self.image_path
        }
